### PR TITLE
Update checkInternalURL to exclude blog pages

### LINKED ISSUE #10721: Shelf module pointing to blog post can trigger 404
We tried publishing a new shelf pointing to one of the new editorial posts on AMO, but ran into some issues. [Here's the shelf](https://addons-internal.prod.mozaws.net/en-US/admin/models/discovery/shelfmodule/7/change/).

The contents of the shelf come from a collection, so the only issue is the "footer" link, which in this case points to `blog/translate-the-web-easily-with-a-browser-extension/`. On the front end, we saw the URL being transformed to `/en-US/firefox/blog/translate-the-web-easily-with-a-browser-extension/`. Now, [that URL](https://addons.mozilla.org/en-US/firefox/blog/translate-the-web-easily-with-a-browser-extension/) works correctly and redirects to the right page if you load it in a new tab, but it seems to glitch between the right page and a 404 if you click on it in the homepage.

I suppose the easier fix here is to not add the `/en-US/firefox/` to blog links, but the other glitch might also be worth investigating.

/cc @devaneymoz @willdurand

### PR DESCRIPTION
**Pull request by bobsilverberg**
Fixes #10721

### CODE REVIEW

**Comment by bobsilverberg** [comment]
@willdurand Andrew won the lottery, but you had an interest in the related issue, so I figured I'd let you know that this PR is up in case you want to take a look.

File: tests/unit/amo/utils/test_index.js [hunk 1]
```diff
@@ -205,6 +205,19 @@ describe(__filename, () => {
           }).isInternal,
         ).toEqual(false);
       });
+
+      it('returns false for a blog URL containing the current host', () => {
+        const baseURL = 'https://example.org';
+        const urlString = url.format({
+          ...url.parse(baseURL),
+          pathname: '/blog/some-blog-post/',
+        });
```
Comment (willdurand) [inline tests/unit/amo/utils/test_index.js:214]: Is this the same as...?

```js
const urlString = `${baseURL}/blog/some-blog-post/`;
```

Comment (bobsilverberg) [reply tests/unit/amo/utils/test_index.js:214]: Pretty much. I just copied an existing test and changed it. At least it maintains some consistency across the tests. 😉

Comment (willdurand) [reply tests/unit/amo/utils/test_index.js:214]: It isn't really clear what's going on with `url.format()` and `pathname`, though. I think having a simple template literal would be a bit better.

File: src/amo/utils/index.js [hunk 2]
```diff
@@ -95,7 +95,12 @@ export const checkInternalURL = ({
   const currentHost = url.parse(baseURL).host || '';
   const isForCurrentHost =
     currentHost === urlParts.host || urlString.startsWith(`//${currentHost}`);
-  const isInternal = isRelative || isForCurrentHost;
+
+  // Treat blog URLs as external.
+  const excludePathRegex = /^\/blog\//;
```
Comment (eviljeff) [inline src/amo/utils/index.js:100]: I feel like external-ish urls should be in a list in config/default.js (like `validLocaleUrlExceptions` which is kinda the same list of AMO urls that aren't served by addons-frontend, but not) but I won't fight for it given it's just one url.

**Review by eviljeff — APPROVED** [review]

File: tests/unit/amo/utils/test_index.js (hunk 1 above)
Comment (bobsilverberg) [reply tests/unit/amo/utils/test_index.js:214]: Sure, I can change that.

File: src/amo/utils/index.js (hunk 2 above)
Comment (bobsilverberg) [reply src/amo/utils/index.js:100]: I had initially thought about it being a list (and maybe in `config`), but, as you say, it's just one path for now, and might continue to be indefinitely.

**Pull request merged**